go/cmd/dolt/commands/sqlserver/serverconfig.go

```diff
@@ -393,6 +426,9 @@ func ValidateConfig(config ServerConfig) error {
 
 // ConnectionString returns a Data Source Name (DSN) to be used by go clients for connecting to a running server.
 func ConnectionString(config ServerConfig) string {
+	if runtime.GOOS != "windows" && config.Host() == "localhost" {
```

Comment:
(minor) This could probably benefit from a helper method to make this more self-descriptive and easier to read. Maybe something like `shouldUseUnixSocket() boolean` or something similar? Seemed like there were a couple of places where that could be used to make the code look a little simpler.